lang/src/org/partiql/lang/eval/Thunk.kt

```diff
@@ -38,8 +40,12 @@ internal typealias ThunkEnv = (Environment) -> ExprValue
  *
  * This name was chosen because it is a thunk that accepts an instance of `Environment` and an [ExprValue] as
  * its arguments.
+ *
+ * @param TEnv The type of the environment.  Generic so that the legacy AST compiler and the new compiler may use
+ * different types here.
+ * @param TArg The type of the additional argument.
```

Comment:
`TArg` doesn't exist. Do you mean `T` here?

Reply:
Meant to rename `T` to `TArg` actually.